### DISCUSSION TITLE
Nested Component Not found

### DISCUSSION THREAD

**Discussion by stescacom** [livewire/livewire#8113] [1. Help]
I have the following setup, and the problem is when trigger `create-sub-item` from the `display-component` it works fine at first, then `edit-sub-item` the `modal` opens just fine, after closing it i get the error `Not found` and on the console  I see error as:

<img width="1318" alt="image" src="https://github.com/livewire/livewire/assets/93092835/2e0965a8-d38d-483c-8edd-be70c0bba653">

I checked the dev tools

<img width="1153" alt="image" src="https://github.com/livewire/livewire/assets/93092835/65aa5f3e-92b4-4eec-bb14-760f2d3c9f8a">

the modal code is below the script tag

```
// page-component.blade.php

<?php

    $data = [
       "name" => "..."
       "sub_items" => [
          [ "id" => "...", "name" => "..." ],
          ...
        ]
   ];

?>

<div>

    <livewire:display-component :data="$data" wire:key="p-d-component" />

    @teleport('body')

        <livewire:form-modal-component :data="$data" wire:key="p-form-component"  />

    @endteleport

</div>



// display-component

<?php
    public $data;

    public function with() {
        return ['items' => $this->data->sub_items];
    }

    #[On('items-updated')]
    public function update_view() {}

?>

<div>
    <button type="button" @click="$dispatch('create-sub-item')">Create</button>
    <ul>
         @foreach($items as $item)
        <li>{{ $item->name }} <button type="button" @click="$dispatch('edit-sub-item', { id: {{ $item->id }} })">Edit</button></li>`
         @endforeach
    </ul>

</div>


// form-modal-component

<?php

public $data;
public $si;

#[On('create-sub-item')]
public function init_create()
{
    // reset..
   $this->dispatch ('open-modal', 'sub-item-form-modal');
}

#[On('edit-sub-item')]
public function init_edit($id)
{
    // reset...
    $this->si = $this->data->sub_items->firstWhere('id', $id);
     // assign data
      $this->dispatch ('open-modal', 'sub-item-form-modal');
}

?>

<x-modal name="sub-item-form-modal" :show="false">
...
</x-modal>

```

**Comment by danie-ramdhani** [comment]
maybe it same case with https://github.com/livewire/livewire/discussions/6993#discussioncomment-8611640, please check your elements.
or maybe it missing the root element in `form-modal-component`

```blade
<div>
    <x-modal name="sub-item-form-modal" :show="false">
    ...
    </x-modal>
</div>
```

**Reply by stescacom** [reply]
With trial and error I have narrowed down the issue to this line

`$this->si = $this->data->sub_items->firstWhere('id', $id);`

**Which I don't understand why!!**
